src/be/mcjava/model/Product.java

```diff
@@ -0,0 +1,70 @@
+package be.mcjava.model;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+/**
+ * Contains all the data related to a single product (e.g. BigMac).
+ * <p>
+ * This class represents the default value for a product, which can later be changed.
+ */
+public class Product {
+    private Long id;
+    private Map<Ingredient, Integer> ingredients;
+    private BigDecimal price;
+    
+    public Product(Map<Ingredient, Integer> ingredients, BigDecimal price) {
+        this(null, ingredients, price);
+    }
+    
+    public Product(Long id, Map<Ingredient, Integer> ingredients, BigDecimal price) {
+        setId(id);
+        setIngredients(ingredients);
+        setPrice(price);
+    }
+    
+    public Long getId() {
+        return id;
+    }
+    
+    public void setId(Long id) {
+        this.id = id;
+    }
+    
+    public Map<Ingredient, Integer> getIngredients() {
+        return ingredients;
+    }
+    
+    public void setIngredients(Map<Ingredient, Integer> ingredients) {
+        this.ingredients = ingredients;
+    }
+    
+    public BigDecimal getPrice() {
+        return price;
+    }
+    
+    public void setPrice(BigDecimal price) {
+        this.price = price;
+    }
+    
+    public boolean isAvailable() {
+        return ingredients.keySet()
+                .stream()
+                .allMatch(Ingredient::isAvailable);
+    }
+    
```

Comment:
You only check if the stock is not null here, you should check if the stock is greater than or equal to the required amount

Reply:
The method now also checks if the stock is greater or equal than the stock.